portal/src/util/misc.ts

```diff
@@ -14,6 +14,21 @@ export function setFieldIfChanged<
   }
 }
 
+export function setNumericFieldIfChanged<M, K extends keyof M>(
+  map: Partial<M>,
+  key: K,
+  initialValue: M[K] extends number ? M[K] : never,
+  value: M[K] extends number ? M[K] : never
```

Comment:
I think `initialValue` and `value` should be string. This avoids the caller to use `Number()` to cast the string into number, which does not always succeed.

Reply:
If the input value is string, then type of `map[key]` will be incompatible with `number`. Fix this by casting to `any` ?

Reply:
After discussion, decided to use `as any` as typescript failed to infer the type from 

```typescript
initialValue: M[K] extends number ? string : never,
value: M[K] extends number ? string : never
```